### PR TITLE
Single mode hover bug #4

### PR DESCRIPTION
**Pull request by rozetko**
Fixed this bug.
Issue: https://github.com/CorpGlory/grafana-discrete-panel/issues/4.

### CODE REVIEW

File: src/module.js [hunk 1]
```diff
@@ -872,7 +878,7 @@ class DiscretePanelCtrl extends CanvasPanelCtrl {
       res.selected = j;
     } else {
       res.items = [j];
-      res.selected = 0;
+      res.selected = j;
```
Comment (rozetko) [inline src/module.js:879]: At single mode returned selected row always was 0, I made it _j_

File: src/module.js [hunk 2]
```diff
@@ -524,21 +524,27 @@ class DiscretePanelCtrl extends CanvasPanelCtrl {
...
       var from = point.start;
       var to = point.start + point.ms;
       var time = point.ms;
       var val = point.val;
-      var seriesName = this.data[i].name;
+      if(!singleMode) {
+        var seriesName = this.data[i].name;
+      } else {
+        var seriesName = this.data[selectedIndex].name;
+      }
```
Comment (rozetko) [inline src/module.js]: At single mode you have just 1 item, therefore name was always _data[0].name_. I changed it to _selectedIndex_.

File: src/module.js [hunk 3]
```diff
@@ -524,21 +524,27 @@ class DiscretePanelCtrl extends CanvasPanelCtrl {
...
 
       var color = this.getColor(val);
 
       body += `
       <div 
         class="
           graph-tooltip-list-item 
-          ${i == selectedIndex ? 'graph-tooltip-list-item--highlight' : ''}
+          ${(i == selectedIndex) || (singleMode) ? 'graph-tooltip-list-item--highlight' : ''}
```
Comment (rozetko) [inline src/module.js]: At single mode you have just 1 item, therefore tooltip line was highlighted only at 1st row.

Comment (jonyrock) [reply src/module.js]: no need to surround with `(`. So `(singleMode)` -> `singleMode`

File: src/module.js [hunk 4]
```diff
@@ -524,21 +524,27 @@ class DiscretePanelCtrl extends CanvasPanelCtrl {
       _.reverse(items);
     }
     
+    var singleMode = (items.length == 1);
```
Comment (jonyrock) [inline src/module.js]: It's a bit strange. 
Single mode should be then it is enabled in options. No?

**Review by jonyrock — CHANGES_REQUESTED** [review]

File: src/module.js (hunk 4 above)
Comment (rozetko) [reply src/module.js]: @jonyrock can you please suggest method to get options? Unfortunately, I couldn't find it.

**Pull request merged**